### PR TITLE
api(matrix): move generateParametrizedTests into test api

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/test-runner/src/cli.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ async function runStage1(command) {
     }
   });
   runner = new Runner(config, new Multiplexer(reporterObjects));
-  if (!runner.loadFiles(testDir, filteredArguments.slice(1), command.testMatch, command.testIgnore).success)
```
Comment (JoelEinbinder) [inline packages/test-runner/src/cli.ts:104]: I put the success property there so I would know whether true or false was good or bad.

Comment (pavelfeldman) [reply packages/test-runner/src/cli.ts:104]: Let's keep it simple - returning true for success is wide spread enough to imply it.

File: packages/test-runner/test/fixtures.ts [hunk 2]
```diff
@@ -112,20 +113,31 @@ fixtures.defineTestFixture('runTest', async ({ outputDir }, testRun, testInfo) =
...
     const { fixtures } = require('${path.join(__dirname, '..').replace(/\\/g, '\\\\')}');
     const { it, expect } = fixtures;
-  `;
+  `, runTest, testRun);
+});
+
+
+fixtures.defineTestFixture('runInlineFixturesTest', async ({ runTest }, testRun) => {
+  await runInlineTest(`
+    const { fixtures: baseFixtures } = require('${path.join(__dirname, '..').replace(/\\/g, '\\\\')}');
```
Comment (JoelEinbinder) [inline packages/test-runner/test/fixtures.ts:126]:
```suggestion
    const { fixtures: baseFixtures } = require(${JSON.stringify(path.join(__dirname, '..')});
```

File: packages/test-runner/test/fixtures.ts [hunk 3]
```diff
@@ -112,20 +113,31 @@ fixtures.defineTestFixture('runTest', async ({ outputDir }, testRun, testInfo) =
...
   await testRun(async files => {
     const dir = await fs.promises.mkdtemp(path.join(tmpdir(), 'playwright-test-runInlineTest'));
     await Promise.all(Object.keys(files).map(async name => {
       await fs.promises.writeFile(path.join(dir, name), header + files[name]);
     }));
-    const {report} = await runTest(dir);
-    await removeFolderAsync(dir);
+    const runResult = await runTest(dir);
+    const { report } = runResult;
+    // await removeFolderAsync(dir);
```
Comment (JoelEinbinder) [inline packages/test-runner/test/fixtures.ts:138]:
```suggestion
    await removeFolderAsync(dir);
```

File: packages/test-runner/test/parameters.spec.ts [hunk 4]
```diff
@@ -34,9 +52,17 @@ it('should run with each configuration', async ({ runTest }) => {
...
 
-it('should fail on invalid parameters', async ({ runTest }) => {
-  const result = await runTest('invalid-parameter');
-  expect(result.exitCode).toBe(1);
-  expect(result.report.parseError.file).toBe(path.join(__dirname, 'assets', 'invalid-parameter', 'setup'));
-  expect(result.report.parseError.error.message).toBe(`Unregistered parameter 'invalid' was set.`);
+it('should fail on invalid parameters', async ({ runInlineTest }) => {
+  const { runResult } = await runInlineTest({
+    'a.spec.ts': `
+      // @ts-ignore
```
Comment (JoelEinbinder) [inline packages/test-runner/test/parameters.spec.ts:58]: Let's not pretend we are type checking these
```suggestion
```

File: packages/test-runner/test/fixtures.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/test-runner/test/fixtures.ts:126]: I'm just moving this code...

File: packages/test-runner/test/fixtures.ts (hunk 3 above)
Comment (pavelfeldman) [reply packages/test-runner/test/fixtures.ts:138]: Ouch, debugging

File: packages/test-runner/test/fixtures.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/test-runner/test/fixtures.ts:126]: Done

File: packages/test-runner/src/cli.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/test-runner/src/cli.ts]: Reverted

**Pull request merged**